src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,47 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+Functional update syntax can only be used on struct types.
+Erroneous code example:
```

Comment:
> I enforced this syntax since it's the most commonly used. Take a look at the RFC.

I prefer "Example of erroneous code" (from the RFC) to "Erroneous code example", although neither sounds very friendly ("erroneous" is a pretty formal sounding construction).

> I generally put it right after the sentence reexplaining the error.

I just meant we need a paragraph break between the summary sentence and the intro to the example.